### DISCUSSION TITLE
SSR for `index.html`

### DISCUSSION THREAD

**Discussion by alexreardon** [parcel-bundler/parcel#4816] [General]
Hi there!

I am currently using `parcel@2.0.0-beta.1` for building a small web app. The `index.html` file is fairly straightforward:

```html
<!DOCTYPE html>
<html lang="en">
  <body>
    <div id="root"></div>
    <script src="./entry.tsx"></script>
  </body>
</html>
```

Then inside of `entry.tsx`:

```tsx
import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';

ReactDOM.render(<App />, document.getElementById('root'));
```

When building for production, I would like to be able to server side render the application. So the new html file would be would be:

```html
<!DOCTYPE html>
<html lang="en">
  <body>
    <div id="root">{result of ReactDOMServer.renderToString() goes here}</div>
    <script src="./entry.tsx"></script>
  </body>
</html>
```

Now doing a hydrate

```tsx
import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';

ReactDOM.hydrate(<App />, document.getElementById('root'));
```

I am exploring a few different ways to do this. Rather than going all over the place I wanted to get your thoughts on what you think the best approach to this is with Parcel

**Comment by mischnic** [comment]
This might be possible with an optimizer plugin that somehow calls `ReactDOMServer.renderToString(....)` and injects it into the html bundle. (Or a packager)

**Reply by alexreardon** [reply]
How could I go about that on my own? I'm thinking for now I could use a seperate entry point for parcel, and the entry point would already have the html rendered into it (so a seperate thing would do reactdom.render. A challenge is that when doing that reactdom.render there are import paths (such as to svgs and css) that node does not understand by default. So I might need to use the parcel bundler to do the reactdom.render
